src/spider/client/Data.hpp

```diff
@@ -0,0 +1,81 @@
+#ifndef SPIDER_CLIENT_DATA_HPP
+#define SPIDER_CLIENT_DATA_HPP
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace spider {
+
+class DataImpl;
+
+template <class T>
+class Data {
+private:
+    std::unique_ptr<DataImpl> m_impl;
+
+public:
+    /**
+     * Gets the values stored in Data.
+     * @return value stored in Data.
+     */
+    auto get() -> T;
+    /**
+     * Indicates that the data is persisted and should not be rollbacked
+     * on failure recovery.
+     */
+    // Not implemented in milestone 1
+    // void mark_persist();
+    /**
+     * Sets locality list of the data.
+     * @param nodes nodes that has locality
+     * @param hard true if the locality list is a hard requirement, false otherwise
```

Comment:
These docstrings are a bit unclear to me. I guess the idea is that if `hard=false`, the data can be accessed from any node in the cluster, but if possible, it should be accessed from the nodes specified in this list?

Reply:
Yes. Hard requirement means data can only be accessed on nodes in the locality list. Will change the docstring to make it clear.